File: java-starter/goodbye-service/src/main/java/org/example/GoodbyeApp.java
package org.example;

import io.nitric.Nitric;

public class GoodbyeApp {
    public static void main(String[] args) {
        var app = Apis.publicApi();

        app.get("/hello/:name", (ctx) -> {
           var name = ctx.getReq().getParams().getOrDefault("name", "world");

           ctx.getResp().text(String.format("Hello %s!", name));

           return ctx;
        });

        Nitric.INSTANCE.run();
    }
}

File: kotlin-starter/goodbye-service/src/main/kotlin/GoodbyeApp.kt
import io.nitric.Nitric
import io.nitric.common.PublicApi

fun main() {
    val api = PublicApi()

    api.get("/goodbye/:name") { ctx ->
        val name = ctx.req.params["name"]

        ctx.resp.text("goodbye $name")

        ctx
    }

    Nitric.run()
}